File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        for i in range(1, len(nums)):
            nums[i] += nums[i-1]
        
        for i in range(len(nums)):
            l = 0 if i == 0 else nums[i-1]
            r = nums[-1] - nums[i]
            if l == r: 
                return i
        
        return -1

File: 0724-find-pivot-index/0724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int[] pfSum = new int[nums.length];
        pfSum[0] = nums[0];
        for (int i = 1; i < nums.length; i++) {
            pfSum[i] = pfSum[i-1] + nums[i];
        }
        
        int s = pfSum[nums.length-1];
        for (int i = 0; i < nums.length; i++) {
            if (pfSum[i] - nums[i] == s - pfSum[i]) return i;
        }
        
        return -1;
    }
}